### ISSUE TITLE
Sqlite uuid relationships

### ISSUE THREAD

**Issue by dvlsg** [typeorm/typeorm#1128]
When using `@PrimaryKeyGenerated('uuid')` for entities with relationships (`@ManyToOne() and @OneToMany()`, specifically), I'm getting errors similar to the following:

```
QueryFailedError: SQLITE_ERROR: foreign key mismatch - "temporary_child" referencing "p
arent"
```

Here's an example which should reproduce the issue, I think. I'm not 100% sure I'm doing it correctly, because I'm new to typeorm, but swapping from `@PrimaryKeyGenerated('uuid')` to just `@PrimaryKeyGenerated()` and a `number` instead of a `string` seems to work.

```ts
import {
  Entity,
  PrimaryGeneratedColumn,
  OneToMany,
  ManyToOne,
  createConnection,
} from 'typeorm'

@Entity()
class Parent {
  @PrimaryGeneratedColumn('uuid') id?: string

  @OneToMany(type => Child, child => child.parent)
  children: Child[]
}

@Entity()
class Child {
  @PrimaryGeneratedColumn('uuid') id?: string

  @ManyToOne(type => Parent, parent => parent.children)
  parent: Parent
}

createConnection({
  type: 'sqlite',
  database: 'test.db',
  entities: [Parent, Child],
  synchronize: true,
}).then(conn => {
  console.log('ready!')
})
```

Here are my versions:

1. typeorm: `0.1.1`
2. sqlite3: `3.1.13`

Haven't tried it with anything other than sqlite just yet. Might try in postgres in a second, if I get a chance.

**Comment by dvlsg** [comment]
Just gave it a shot with postgres (v10, I think), seems to work fine.

**Comment by maddindeiss** [comment]
Hi, 
I'm getting a similar error here with the `cordova` driver.

`sqlite3_prepare_v2 failure: foreign key mismatch - "temporary_project" referencing "user"`

Example code:

``` javascript
@Entity('project')
export class Project {
  @PrimaryGeneratedColumn('uuid')
  public id: string;

  @ManyToOne(type => User)
  public user: User
}

@Entity('user')
export class User {
  @PrimaryGeneratedColumn('uuid')
  public id: string;

  @Column({ name: 'name', type: 'varchar', length: 255 })
  public name: string
}

createConnection({
  type: 'cordova',
  location: 'default',
  database: 'test',
  entities: [
    Project, 
    User
  ],
  synchronize: true,
}).then(conn => {
  console.log('ready!')
})
```

I'm using:  
- ionic `3.8.0`
- typeorm: `0.1.2`
- cordova-sqlite-storage: `2.1.1`

**Comment by benedictkhoo** [comment]
Hi,

I too am getting the same error using sqlite.

I enabled logging and I noticed that my entities with `@PrimaryGeneratedColumn('uuid')` does not generate a SQL statement using `CREATE TABLE` with `PRIMARY KEY(id)`.

With `@PrimaryColumn('uuid')`:

```typescript
@Entity()
export class User {
  @PrimaryColumn('uuid')
  id: string;
}
```

```sql
CREATE TABLE "user" ("id" varchar NOT NULL, PRIMARY KEY(id))
```

With `@PrimaryGeneratedColumn('uuid')`:

```typescript
@Entity()
export class Message {
  @PrimaryGeneratedColumn('uuid')
  id: string;
}
```

```sql
CREATE TABLE "message" ("id" varchar NOT NULL)
```

I'm using:

- typeorm: `0.1.1`
- sqlite3: `3.1.13`

**Comment by pleerock** [comment]
This issue is fixed and released in `0.1.6`.
